server/models/base/Model.ts

```diff
@@ -3,13 +3,158 @@ import isEqual from "fast-deep-equal";
 import isArray from "lodash/isArray";
 import isObject from "lodash/isObject";
 import pick from "lodash/pick";
-import { FindOptions, NonAttribute } from "sequelize";
-import { Model as SequelizeModel } from "sequelize-typescript";
+import {
+  CreateOptions,
+  CreationAttributes,
+  FindOptions,
+  InstanceDestroyOptions,
+  InstanceUpdateOptions,
+  ModelStatic,
+  NonAttribute,
+} from "sequelize";
+import {
+  AfterCreate,
+  AfterDestroy,
+  AfterUpdate,
+  Model as SequelizeModel,
+} from "sequelize-typescript";
+import Logger from "@server/logging/Logger";
+import { APIContext } from "@server/types";
 
 class Model<
   TModelAttributes extends {} = any,
   TCreationAttributes extends {} = TModelAttributes
 > extends SequelizeModel<TModelAttributes, TCreationAttributes> {
+  /**
+   * The namespace to use for events, if none is provided the table name is used.
+   */
+  static eventNamespace: string | undefined;
+
+  /**
+   * The properties to include in the event data when this model is mutated.
+   */
+  static eventData: string[] = [];
```

Comment:
`changeset` was added since this PR started, so I think now we can remove this and just record the full changeset for every event. Although there will be some exceptions, like with documents the full content/state should not be stored.

Maybe a method on model to `getEventData` that defaults to the changeset and can be overridden by other models is the way to go.

Reply:
Yup, adds the much needed type-safety as well!
Other than this, I didn't find anything of concern tbh.